### PR TITLE
Upgrade/application api connection

### PR DESCRIPTION
**Pull request by mmumby**
## Main Updates
- removed all mocks for applications
- Connected School application to API (GET by Id and GET all)
- Updated all fields to pickLists on school application form
- updated a few read only fields

### Other work
- Renamed EOI status' to remove additional words on main table view
- Removed profile link from UI/Commented out profile route for now

### CODE REVIEW

File: frontend/src/store/modules/applications.js [hunk 1]
```diff
@@ -1,26 +1,43 @@
...
     getSchoolApplicationsFormatted: (state) =>
-      // format the data for table view - keys are turned into table headers
-      Object.values(Object.fromEntries(state.schoolApplicationsMap)).map(
-        (v) => ({
-          // TODO: update with id when connected to API
+      // Sort applications by last 4 digits of the applicationnumber
+      state.schoolApplications
+        .sort(
+          (a, b) =>
+            a.iosas_applicationnumber.split('-').pop() -
```
Comment (fwpushan) [inline frontend/src/store/modules/applications.js]: Can we use a var name with more than 3 letters?

File: frontend/src/store/modules/applications.js [hunk 2]
```diff
@@ -40,8 +56,9 @@ export const applicationsStore = defineStore('applications', {
           b.iosas_eoinumber.split('-').pop()
       ).map((v) => ({
```
Comment (fwpushan) [inline frontend/src/store/modules/applications.js]: same as var naming

**Review by fwpushan — COMMENTED** [review]
All good, please check some minor comments.

File: frontend/src/store/modules/applications.js (hunk 1 above)
Comment (mmumby) [reply frontend/src/store/modules/applications.js]: Updated!

File: frontend/src/store/modules/applications.js (hunk 2 above)
Comment (mmumby) [reply frontend/src/store/modules/applications.js]: updated!

**Review by fwpushan — APPROVED** [review]

**Pull request merged**